modules/diskq/logqueue-disk.c

```diff
@@ -194,6 +194,28 @@ log_queue_disk_restart_corrupted(LogQueueDisk *self)
   _restart_diskq(self);
 }
 
+static gboolean
+_serialize_msg_with_compaction(LogMessage *msg, SerializeArchive *sa)
+{
+  return log_msg_serialize(msg, sa, LMSF_COMPACTION);
+}
+
+static gboolean
+_serialize_msg_without_compaction(LogMessage *msg, SerializeArchive *sa)
+{
+  return log_msg_serialize(msg, sa, 0);
+}
+
+static void
+_set_serializer(LogQueueDisk *self)
+{
+  DiskQueueOptions *options = qdisk_get_options(self->qdisk);
+
+  if (options->compaction)
+    self->serialize_msg = _serialize_msg_with_compaction;
+  else
+    self->serialize_msg = _serialize_msg_without_compaction;
```

Comment:
Currently there is only the compaction flag exists here, which makes this a simple condition and two function, but possible adding more flags would make it less appealing to branch out due to these flags. (In addition this replaces a simple int argument passing with pointer dereference.)

Reply:
The serialize_msg function needs a LogMessaget and a SerializeArchive parameter (and nothing else) for the save_queue function to get the serialize function. If the save_queue function does not need the serialize function, then can get the flag as a parameter.

So in the future can be an int parameter, but right now can not

Reply:
I was also curious why we replaced a simple condition with a pointer dereference. I thought it was a virtual method for `LogQueueDisk` that can be overridden, but it turned out that we have another layering problem in `QDisk` (`save_queue()`), which we could fix later. This function pointer temporarily keeps `save_queue()` in a functioning state without requiring a `LogQueueDisk` or a `DiskQueueOptions` instance with multiple LogQueueDisk calls inside QDisk.